src/main/java/com/hivemq/client/mqtt/mqtt5/message/publish/Mqtt5Publish.java

```diff
@@ -110,6 +110,15 @@
      */
     @NotNull Mqtt5UserProperties getUserProperties();
 
+    /**
+     * Acknowledges this Publish message.
+     *
+     * @throws UnsupportedOperationException if manual acknowledgement is not enabled.
+     * @throws IllegalStateException         if the message is acknowledged more than once.
```

Comment:
Why not also if its a qos 0 message?

Reply:
This decision was made in the hivemq-mqtt-client slack channel, and the reason is that it is a programming error if it is called multiple times.

Reply:
Good that I was not in the channel at the time. I would have argued both times in the opposite direction. ;)

Reply:
If you have good reasons we can have another discussion on slack.